### PR TITLE
Fiducial registration, ambiguous point mapping warning

### PR DESCRIPTION
**Pull request by tavaughan**

### CODE REVIEW

**Comment by tavaughan** [comment]
@lassoan Would you like to review, or should I go ahead and merge?

File: FiducialRegistrationWizard/Logic/vtkPointMatcher.cxx [hunk 1]
```diff
@@ -320,20 +342,25 @@ void vtkPointMatcher::UpdateBestMatchingForSubsetOfPoints( vtkPoints* pointSubse
...
     double rootMeanSquareDistanceErrorMm = this->ComputeRootMeanSquareistanceErrors( pointSubset1DistanceMatrix, permutedPointSubset2DistanceMatrix );
 
-    // TODO
-    // if suitability within some threshold of the best
-    //   flag ambiguous
+    // flag ambiguous if suitability within some threshold of the best result so far
+    double differenceComparedToBestMm = this->ComputedRootMeanSquareDistanceErrorMm - rootMeanSquareDistanceErrorMm;
+    if ( fabs( differenceComparedToBestMm ) <= this->AmbiguityThresholdDistanceMm )
+    {
+      this->MatchingAmbiguous = true;
```
Comment (lassoan) [inline FiducialRegistrationWizard/Logic/vtkPointMatcher.cxx]: Is IsMatchingAmbiguous variable guaranteed to be set in all cases?

File: FiducialRegistrationWizard/Logic/vtkPointMatcher.h [hunk 2]
```diff
@@ -63,6 +66,17 @@ class VTK_SLICER_FIDUCIALREGISTRATIONWIZARD_MODULE_LOGIC_EXPORT vtkPointMatcher
     // The mean distance error between each pair of points, after matching
     double ComputedRootMeanSquareDistanceErrorMm;
 
+    // Determine if there is more than one plausible mapping.
```
Comment (lassoan) [inline FiducialRegistrationWizard/Logic/vtkPointMatcher.h]: Not clear what mapping you mean.

File: FiducialRegistrationWizard/Logic/vtkPointMatcher.cxx (hunk 1 above)
Comment (tavaughan) [reply FiducialRegistrationWizard/Logic/vtkPointMatcher.cxx]: What do you mean by "guaranteed to be set"? What do you mean by "all cases"?

File: FiducialRegistrationWizard/Logic/vtkPointMatcher.h (hunk 2 above)
Comment (tavaughan) [reply FiducialRegistrationWizard/Logic/vtkPointMatcher.h]: Would it be clearer if I said point-to-point mapping?

Comment (lassoan) [reply FiducialRegistrationWizard/Logic/vtkPointMatcher.h]: Somewhat better. It is still not clear what point to what point. Something like "reordering of from/to markup points for best match of positions" would be more clear.

File: FiducialRegistrationWizard/Logic/vtkPointMatcher.cxx (hunk 1 above)
Comment (lassoan) [reply FiducialRegistrationWizard/Logic/vtkPointMatcher.cxx]: Just make sure that you always assign a value to this->MatchingAmbiguous. Currently, it is not obvious that it is set in all cases, only when you get inside one of the if statements.

**Comment by tavaughan** [comment]
@lassoan Not sure if you want to look over changes, or if I should just push to master?

**Comment by lassoan** [comment]
You can push trivial fixes or things that you are completely confident about (documentation, comments, copy-paste or other obvious errors) directly to master. If you want to have something looked at before you merge then send a pull request instead. If in doubt, send a pull request.

**Pull request merged**